### PR TITLE
fix(withForm): use functional setState

### PR DESCRIPTION
**Pull request by dadamssg**

### CODE REVIEW

File: src/withForm.js [hunk 1]
```diff
@@ -9,37 +9,42 @@ const withForm = (BaseComponent) => {
...
         initialValues,
         currentValues: currentValues ? fromJS(currentValues) : initialValues
-      })
+      }))
     }
     setValue = (field, value) => {
-      this.setState({
+      this.setState((state) =>  ({
+        ...state,
         currentValues: this.state.currentValues.set(field, fromJS(value))
```
Comment (donovanCarreno) [inline src/withForm.js:21]: should this be `state.` instead of `this.state.`?

Comment (dadamssg) [reply src/withForm.js]: good catch

**Comment by dadamssg** [comment]
@donovanCarreno look good now?

**Pull request merged**